llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -3421,6 +3421,32 @@ Value *InstCombinerImpl::foldAndOrOfICmps(ICmpInst *LHS, ICmpInst *RHS,
   return foldAndOrOfICmpsUsingRanges(LHS, RHS, IsAnd);
 }
 
+static Value *foldAorBConst(BinaryOperator &I,
+                            InstCombiner::BuilderTy &Builder) {
+  assert(I.getOpcode() == Instruction::Or &&
+         "Simplification only supports or at the moment.");
+
+  Value *Op0 = I.getOperand(0);
+  Value *Op1 = I.getOperand(1);
+
+  Value *Cmp1, *Cmp2, *Cmp3, *Cmp4;
+  if (!match(Op0, m_And(m_Value(Cmp1), m_Value(Cmp2))) ||
+      !match(Op1, m_And(m_Value(Cmp3), m_Value(Cmp4))))
```

Comment:
```suggestion
  if (!match(I.getOperand(0), m_And(m_Value(Cmp1), m_Value(Cmp2))) ||
      !match(I.getOperand(1), m_And(m_Value(Cmp3), m_Value(Cmp4))))
```
No real need to introduce Op0 and Op1 variables to use them in a single place...